fw/charybdis-features.md

```diff
@@ -56,72 +58,115 @@ For each mode, the firmware allows cycling through multiple pre-defined values:
 
-{: .note }
-This requires a firmware update.
 
-The following section explains how to update the firmware source code to fine-tune your experience. This requires a functional development environment for QMK (https://docs.qmk.fm/#/newbs) and, ideally, some prior experience in computer science and the C programming language.
+Custom defines (with default values):
 
-### Changing dynamic DPI scaling default and increment values
+```c
+#define CHARYBDIS_MINIMUM_SNIPING_DPI 200
+#define CHARYBDIS_SNIPING_DPI_CONFIG_STEP 100
+```
 
-{: .note }
-This requires a firmware update.
+Custom functions:
+
+```c
+charybdis_set_pointer_sniping_enabled(bool enable) // enable/disable sniping mode
+charybdis_get_pointer_sniping_enabled() // returns whether sniping mode is currently enabled
+charybdis_cycle_pointer_sniping_dpi(bool forward) // cycle forward or backward the possible values
+charybdis_cycle_pointer_sniping_dpi_noeeprom(bool forward) // cycle forward or backward the possible values without persisting the change to EEPROM
+charybdis_get_pointer_sniping_dpi() // returns the current sniping mode DPI value
+```
 
 ### Auto sniping on layer
 
-{: .note }
-This requires a firmware update.
+You can trigger sniping automatically when on a specific layer by adjusting the following in your keymap:
+
+```c
+#define CHARYBDIS_AUTO_SNIPING_ON_LAYER LAYER_POINTER
+```
 
 ### Auto pointer layer
 
-{: .note }
-This requires a firmware update.
+You can trigger the pointer layer automatically upon moving the trackball by adjusting the following in your keymap:
+
+```c
+#define CHARYBDIS_AUTO_POINTER_LAYER_TRIGGER_ENABLE
+#define CHARYBDIS_AUTO_POINTER_LAYER_TRIGGER_TIMEOUT_MS 1000
+```
+
+### Drag-scroll
+
+**Drag-scroll** enables scrolling with the trackball. When drag-scroll is enabled, the trackball's `x` and `y` movements are converted into `h` (horizontal) and `v` (vertical) movement, effectively sending scroll instructions to the host system.
+
+Custom keycodes:
+
+| Name   | Description                                           |
+| ------ | ----------------------------------------------------- |
+| `DRGSCRL`  | enable drag-scroll mode as long as the key is pressed |
+| `DRG_TOG` | toggle drag-scroll mode on and off                    |
+
+Custom functions:
+
+```c
+charybdis_set_pointer_dragscroll_enabled(bool enable) // enable/disable drag-scroll
+charybdis_get_pointer_dragscroll_enabled() // returns whether drag-scroll mode is currently enabled
+```
+
+Custom defines:
+
+```c
+#define CHARYBDIS_DRAGSCROLL_REVERSE_X` // inverts horizontal scrolling 
+#define CHARYBDIS_DRAGSCROLL_REVERSE_Y` // inverts vertical scrolling 
+```
+
+### Large mouse reports
 
-### X/Y axis inversion
+By default, the `x` and `y` motion for the pointing device/mouse reports is `-127` to `127`. If you hit the limit for that with the sensors, you can enable support for `-32767` to `32767` by adding this to your `config.h`:
```

Comment:
Is this necessary even at max dpi?

Reply:
no idea about this, I'm not technical enough. We could just remove it?

Reply:
You could either remove it or try setting a charybdis to max dpi (12000) and see if the cursor skips/warps. If it doesn't, its not necessary.